File: sample/app/src/main/java/me/shouheng/samples/ui/TestAnimUtilsActivity.java
package me.shouheng.samples.ui;

import android.graphics.Color;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import me.shouheng.samples.R;
import me.shouheng.utils.ui.AnimUtils;
import me.shouheng.utils.ui.ViewUtils;

/**
 * Anim utils test
 *
 * @author <a href="mailto:shouheng2015@gmail.com">WngShhng</a>
 * @version 2020-05-24 12:53
 */
public class TestAnimUtilsActivity extends AppCompatActivity {

    private View obj;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_anim_utils);
        obj = findViewById(R.id.v);
        findViewById(R.id.btn_fade_in).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                AnimUtils.fadeIn(obj, 2000, null, true);
            }
        });
        findViewById(R.id.btn_fade_out).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                AnimUtils.fadeOut(obj, 2000, null, true);
            }
        });
        findViewById(R.id.btn_slide_in).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                AnimUtils.slideIn(obj, 2000, null, true, AnimUtils.UIDirection.RIGHT_TO_LEFT);
            }
        });
        findViewById(R.id.btn_slide_out).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                AnimUtils.slideOut(obj, 2000, null, true, AnimUtils.UIDirection.LEFT_TO_RIGHT);
            }
        });
        findViewById(R.id.btn_shinning).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                AnimUtils.shining(obj, 2888, 6, 0, 0.66f, 1.0f, 0);
            }
        });
    }

    public void shake(View view) {
        AnimUtils.shake(obj);
    }

    public void changeColor(View view) {
        AnimUtils.changeColor(Color.RED, Color.BLUE, 3000, new AnimUtils.OnColorChangeListener() {
            @Override
            public void onColorChanged(int color) {
                obj.setBackgroundColor(color);
            }
        });
    }

    public void onZoomIn(View view) {
        AnimUtils.zoomIn(obj, .5f, 0f, 300);
    }

    public void onZoomOut(View view) {
        AnimUtils.zoomOut(obj, .5f, 300);
    }

    public void scaleUpDown(View view) {
        AnimUtils.scaleUpDown(obj, 1200);
    }

    public void animateHeight(View view) {
        AnimUtils.animateHeight(obj, 0, ViewUtils.dp2px(100f));
    }

    public void popupIn(View view) {
        AnimUtils.popupIn(obj, 1000);
    }

    public void popupOut(View view) {
        AnimUtils.popupOut(obj, 1000, null);
    }
}

File: sample/app/src/main/java/me/shouheng/samples/ui/TestAnimUtilsActivity.kt
package me.shouheng.samples.ui

import android.graphics.Color
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.View
import me.shouheng.samples.R
import me.shouheng.utils.ktx.*
import me.shouheng.utils.ui.AnimUtils
import me.shouheng.utils.ui.ViewUtils
import android.view.animation.Animation.RELATIVE_TO_SELF

/**
 * Anim utils test
 *
 * @author [Shouheng Wang](mailto:shouheng2020@gmail.com)
 * @version 2020-05-24 12:53
 */
class TestAnimUtilsActivity : AppCompatActivity() {
    private var obj: View? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_anim_utils)
        obj = findViewById(R.id.v)
        findViewById<View>(R.id.btn_fade_in).onDebouncedClick   { obj?.fadeIn(2000, null, true) }
        findViewById<View>(R.id.btn_fade_out).onDebouncedClick  { obj?.fadeOut(2000, null, true) }
        findViewById<View>(R.id.btn_slide_in).onDebouncedClick  { obj?.slideIn(2000, null, true, AnimUtils.UIDirection.RIGHT_TO_LEFT) }
        findViewById<View>(R.id.btn_slide_out).onDebouncedClick { obj?.slideOut(2000, null, true, AnimUtils.UIDirection.LEFT_TO_RIGHT) }
        findViewById<View>(R.id.btn_shinning).onDebouncedClick  { obj?.shining(2888, 6, 0f, 0.66f, 1.0f, 0f) }
    }

    fun shake(view: View?) {
        obj?.shake()
    }

    fun shakeUpDown(view: View?) {
        obj?.shake(AnimUtils.UIDirection.BOTTOM_TO_TOP)
    }

    fun changeColor(view: View?) {
        obj?.changeBackgroundColor(Color.RED, Color.BLUE, 3000)
//        changeColor(Color.RED, Color.BLUE, 3000) { obj!!.setBackgroundColor(it) }
//        AnimUtils.changeColor(Color.RED, Color.BLUE, 3000) {  }
    }

    fun onZoomIn(view: View?) {
        obj?.zoomIn(.5f, 0f, 300)
    }

    fun onZoomOut(view: View?) {
        obj?.zoomOut(.5f, 300)
    }

    fun scaleUpDown(view: View?) {
        obj?.scaleUpDown(1200)
    }

    fun animateHeight(view: View?) {
        obj?.animateHeight(0, ViewUtils.dp2px(100f))
    }

    fun popupIn(view: View?) {
        obj?.popupIn(1000)
    }

    fun popupOut(view: View?) {
        obj?.popupOut(1000, null)
    }

    fun rotate(view: View) {
        obj?.rotate(2_000)
    }

    fun scale(view: View) {
        obj?.scales(500L, 1f, .5f, 1f)
    }

    fun scales(view: View) {
        obj?.scales(500L, .5f, RELATIVE_TO_SELF, 1f, .5f, 1.5f, 0.5f, 1f)
    }
}